agent/server.go

```diff
@@ -0,0 +1,142 @@
+package agent
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"log"
+	"net/http"
+	"os"
+	"time"
+
+	"github.com/cybozu-go/well"
+	"github.com/go-logr/logr"
+	"github.com/slack-go/slack"
+	"github.com/slack-go/slack/socketmode"
+)
+
+const (
+	resultAPIPath    = "/result"
+	slackPostTimeout = 3 * time.Second
+)
+
+type resultAPIPayload struct {
+	Color   string `json:"color"`
+	Text    string `json:"text"`
+	Job     string `json:"job"`
+	Pod     string `json:"pod"`
+	Extend  bool   `json:"extend"`
+	Channel string `json:"channel"`
+}
+
+// Server receives requests from clients and communicates with Slack.
+type Server struct {
+	log            logr.Logger
+	listenAddr     string
+	defaultChannel string
+	apiClient      *slack.Client
+	smClient       *socketmode.Client
```

Comment:
Can you use mocks (apiClient, smClient and extendPod) to write unit tests?

Reply:
In my opinion, this unit tests are sufficient.
https://github.com/cybozu-go/github-actions-controller/blob/job-cancel/agent/socket_test.go